trace/LogCorrelation.md

```diff
@@ -0,0 +1,70 @@
+# Log Correlation
+
+Log correlation is a feature that inserts information about the current span into log entries
+created by existing logging frameworks.  The feature can be used to add more context to log entries,
+filter log entries by trace ID, or show log entries as annotations on a trace.
+
+The design of a log correlation implementation depends heavily on the details of the particular
+logging framework that it supports.  Therefore, this document only covers the aspects of log
+correlation that could be shared across log correlation implementations for multiple languages and
+logging frameworks.  It doesn't cover how to hook into the logging framework.
+
+## Identifying the span to associate with a log entry
+
+A log correlation implementation should look up tracing data from the span that is current at the
+point of the log statement.  See
+[Span.md#how-span-interacts-with-context](Span.md#how-span-interacts-with-context) for the
+definition of the current span.
+
+## Tracing data to include in log entries
+
+A log correlation implementation should insert the following pieces of tracing data from the current
+span context into each log entry:
+
+### Trace ID
+
+The trace ID of the current span.  See [Span.md#traceid](Span.md#traceid).
+
+### Span ID
+
+The span ID of the current span.  See [Span.md#spanid](Span.md#spanid).
+
+### Sampling Decision
+
+The sampling bit of the current span, as a boolean.  See
+[Span.md#supported-bits](Span.md#supported-bits).
+
+## String format for tracing data
+
+The logging framework may require the pieces of tracing data to be converted to strings.  In that
+case, the log correlation implementation should format the trace ID and span ID as lowercase base 16
+and format the sampling decision as "true" or "false".
+
+## Key names for tracing data
+
+Some logging frameworks allow the insertion of arbitrary key-value pairs into log entries.  When
+a log correlation implementation inserts tracing data by that method, the key names should be
+"opencensusTraceId", "opencensusSpanId", and "opencensusTraceSampled" by default.  The log
+correlation implementation may allow the user to override the tracing data key names.
+
+## Deciding when to add tracing data to a log entry
```

Comment:
I think this feature is a bit speculative. I would leave it out as I have only seen this practice here. usually it isnt selective.